# MIT License
# 
# Copyright (c) 2016 AUTHORS
# 
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
# 
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
# 
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.diff -cr -N -x .git -x '*.patch' /old/AUTHORS /new/AUTHORS
*** /old/AUTHORS	2016-12-22 12:17:26.820824853 -0500
--- /new/AUTHORS	2016-12-21 19:15:05.011980080 -0500
***************
*** 1,2 ****
! isysd ira@gitguild.com 5C3586F6
  
--- 1,2 ----
! <<< user_name >>> <<< user_email >>> <<< user_signingkey >>>
  
diff -cr -N -x .git -x '*.patch' /old/CHANGELOG.md /new/CHANGELOG.md
*** /old/CHANGELOG.md	2016-12-22 12:17:26.812824853 -0500
--- /new/CHANGELOG.md	2016-12-21 19:15:04.975980081 -0500
***************
*** 7,13 ****
  <https://github.com/olivierlacan/keep-a-changelog/blob/master/CHANGELOG.md>
  
  <a name="0.0.1"></a>
! ## 0.0.1 (2016-12-22)
  
  
  #### Features
--- 7,13 ----
  <https://github.com/olivierlacan/keep-a-changelog/blob/master/CHANGELOG.md>
  
  <a name="0.0.1"></a>
! ## 0.0.1 (<<< date >>>)
  
  
  #### Features
diff -cr -N -x .git -x '*.patch' /old/GUILD /new/GUILD
*** /old/GUILD	2016-12-22 12:17:26.824824853 -0500
--- /new/GUILD	2016-12-21 19:15:05.011980080 -0500
***************
*** 1,5 ****
! NAME isysd
! HEIGHT 1
! LAST_TRANSACTION 
! EPOCH_TIME 1482425327
  
--- 1,5 ----
! NAME <<< guild_name >>>
! HEIGHT <<< height >>>
! LAST_TRANSACTION <<< last_transaction >>>
! EPOCH_TIME <<< epoch >>>
  
diff -cr -N -x .git -x '*.patch' /old/ledger/account.ledger /new/ledger/account.ledger
*** /old/ledger/account.ledger	2016-12-22 12:17:26.812824853 -0500
--- /new/ledger/account.ledger	2016-12-21 19:15:04.975980081 -0500
***************
*** 8,44 ****
  ; User account also has two additional options related to voting: Height, Depth.
  ; Metadata might be bitcoin or dash addresses, and/or transaction hashes.
  
! account isysd
      note The Guild's own account.
! account isysd:Expenses
      note The Guild's Expenses, i.e. Asset redemptions, contribution pay origination
  
! account isysd:Depth
      note the Guild's voting account. Votes in XP the amount of qualified Income earned by member(s).
      assert commodity == "XP"
! account isysd:Height
      note the Guild's vote counting area. Votes by members on commits related to this guild are recorded here.
      assert commodity == "XP"
! account isysd:Equity
! account isysd:Income
! account isysd:Assets
! account isysd:Liabilities
      note This guild can only have XGG liabilities, and this is the source of all XGG.
! account isysd:Expenses
      note Members earning Income on approved labor can ask to cash in XGG against the Guild's Assets, creating an Expense
  
! account isysd:Depth
!     note The votes that isysd has spent. Cannot spend more than account height on a commit/issue.
! account isysd:Height
!     note The number of votes that isysd has received from the guild and also the XP voting power of isysd.
  ; users do not get Equity as members of a shared guild.
! account isysd:Income
! account isysd:Assets
! account isysd:Liabilities
! account isysd:Expenses
  
! account isysd:Equity:init:f5cff1bebf664c244d68442f045939622bc2a5dd
  
! account isysd:Equity:register:f5cff1bebf664c244d68442f045939622bc2a5dd
  
  
--- 8,44 ----
  ; User account also has two additional options related to voting: Height, Depth.
  ; Metadata might be bitcoin or dash addresses, and/or transaction hashes.
  
! account <<< guild_name >>>
      note The Guild's own account.
! account <<< guild_name >>>:Expenses
      note The Guild's Expenses, i.e. Asset redemptions, contribution pay origination
  
! account <<< guild_name >>>:Depth
      note the Guild's voting account. Votes in XP the amount of qualified Income earned by member(s).
      assert commodity == "XP"
! account <<< guild_name >>>:Height
      note the Guild's vote counting area. Votes by members on commits related to this guild are recorded here.
      assert commodity == "XP"
! account <<< guild_name >>>:Equity
! account <<< guild_name >>>:Income
! account <<< guild_name >>>:Assets
! account <<< guild_name >>>:Liabilities
      note This guild can only have XGG liabilities, and this is the source of all XGG.
! account <<< guild_name >>>:Expenses
      note Members earning Income on approved labor can ask to cash in XGG against the Guild's Assets, creating an Expense
  
! account <<< user_name >>>:Depth
!     note The votes that <<< user_name >>> has spent. Cannot spend more than account height on a commit/issue.
! account <<< user_name >>>:Height
!     note The number of votes that <<< user_name >>> has received from the guild and also the XP voting power of <<< user_name >>>.
  ; users do not get Equity as members of a shared guild.
! account <<< user_name >>>:Income
! account <<< user_name >>>:Assets
! account <<< user_name >>>:Liabilities
! account <<< user_name >>>:Expenses
  
! account <<< user_name >>>:Equity:init:<<< parent >>>
  
! account <<< user_name >>>:Equity:register:<<< parent >>>
  
  
diff -cr -N -x .git -x '*.patch' /old/ledger/check.ledger /new/ledger/check.ledger
*** /old/ledger/check.ledger	2016-12-22 12:17:26.812824853 -0500
--- /new/ledger/check.ledger	2016-12-21 19:15:04.975980081 -0500
***************
*** 1,8 ****
  ; TODO lots of assertions and checks go here to ensure proper behavior
  ; these will be suplimented with file templates and transformations
  
! define GUILD_DEPTH = (-account("isysd:Depth:isysd").total)
! define GUILD_HEIGHT = (account("isysd:Height:isysd").total)
  
  ; Each user must have net positive score
  define user_vote_check(user_name) = (account(user_name + ":Height").total >= GUILD_DEPTH / 2)
--- 1,8 ----
  ; TODO lots of assertions and checks go here to ensure proper behavior
  ; these will be suplimented with file templates and transformations
  
! define GUILD_DEPTH = (-account("<<< guild_name >>>:Depth:<<< guild_name >>>").total)
! define GUILD_HEIGHT = (account("<<< guild_name >>>:Height:<<< guild_name >>>").total)
  
  ; Each user must have net positive score
  define user_vote_check(user_name) = (account(user_name + ":Height").total >= GUILD_DEPTH / 2)
***************
*** 17,26 ****
  define user_issue_vote_check(user_name, vote_guild_name, vote_branch, vote_parent) = (account(user_name + ":Height").total >= abs(account(user_name + ":Depth:" + vote_guild_name + ":" + vote_branch + ":" + vote_parent).total))
  
  ; Guild itself must have positive height
! check user_vote_check("isysd")
  
! check branch_vote_check("isysd", "isysd", "isysd")
! check issue_vote_check("isysd", "isysd", "isysd", "f5cff1bebf664c244d68442f045939622bc2a5dd")
! check branch_vote_check("isysd", "isysd", "isysd")
! check user_issue_vote_check("isysd", "isysd", "isysd", "f5cff1bebf664c244d68442f045939622bc2a5dd")
  
--- 17,26 ----
  define user_issue_vote_check(user_name, vote_guild_name, vote_branch, vote_parent) = (account(user_name + ":Height").total >= abs(account(user_name + ":Depth:" + vote_guild_name + ":" + vote_branch + ":" + vote_parent).total))
  
  ; Guild itself must have positive height
! check user_vote_check("<<< guild_name >>>")
  
! check branch_vote_check("<<< guild_name >>>", "<<< guild_name >>>", "<<< guild_name >>>")
! check issue_vote_check("<<< guild_name >>>", "<<< guild_name >>>", "<<< guild_name >>>", "<<< vote_parent >>>")
! check branch_vote_check("<<< guild_name >>>", "<<< guild_name >>>", "<<< guild_name >>>")
! check user_issue_vote_check("<<< guild_name >>>", "<<< guild_name >>>", "<<< guild_name >>>", "<<< vote_parent >>>")
  
diff -cr -N -x .git -x '*.patch' /old/ledger/transaction.ledger /new/ledger/transaction.ledger
*** /old/ledger/transaction.ledger	2016-12-22 12:17:26.812824853 -0500
--- /new/ledger/transaction.ledger	2016-12-21 19:15:04.975980081 -0500
***************
*** 1,6 ****
  
! 2016-12-22 * isysd Personal Guild Initialization
      ; Create XP by giving the founder Equity
!     isysd:Height:isysd:isysd:f5cff1bebf664c244d68442f045939622bc2a5dd    1 XP
!     isysd:Depth:isysd:isysd:f5cff1bebf664c244d68442f045939622bc2a5dd    -1 XP
  
--- 1,6 ----
  
! <<< date >>> * <<< user_name >>> Personal Guild Initialization
      ; Create XP by giving the founder Equity
!     <<< user_name >>>:Height:<<< user_name >>>:<<< user_name >>>:<<< parent >>>    1 XP
!     <<< user_name >>>:Depth:<<< user_name >>>:<<< user_name >>>:<<< parent >>>    -1 XP
  
diff -cr -N -x .git -x '*.patch' /old/VERSION /new/VERSION
*** /old/VERSION	2016-12-22 12:17:26.820824853 -0500
--- /new/VERSION	2016-12-22 11:36:33.260868791 -0500
***************
*** 1 ****
! 
--- 1 ----
! <<< version >>>
